GOT-StudentVersion/GOT-StudentVersion/ViewController.swift

```diff
@@ -8,18 +8,144 @@
 
 import UIKit
 
-class ViewController: UIViewController {
-
+class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    @IBOutlet weak var tableView: UITableView!
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let test = Functions()
+        return test.rows(section)
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch indexPath.section{
```

Comment:
This function is asking for the data each cell holds. You hardcoded each individual case knowing how many seasons you have, but what if you didn't know how many seasons you have or what if a season gets added or removed from the model? That's why you make an instance of the model. That way, no matter what changes you already hold that.
Here you can refactor this code like: 
 let episode = arrOfSeasons[indexPath.section][indexPath.row]
        var resultCell = UITableViewCell()
        if episode.season % 2 == 0 {
            guard let cell = tableView.dequeueReusableCell(withIdentifier: "GameThronesCellTwo", for: indexPath) as? CustomCell else {fatalError("GOT cell not found")}
            
            cell.GOTImage.image = UIImage(named: episode.mediumImageID)
            cell.episodeName.text = episode.name
            cell.seasonEpisode.text = "S\(episode.season) E: \(episode.number)"
            resultCell = cell
        } else if episode.season % 2 != 0 {
            guard let cell = tableView.dequeueReusableCell(withIdentifier: "GOTcell", for: indexPath) as? CustomCell else {fatalError("GOT cell not found")}
            
            cell.GOTImage.image = UIImage(named: episode.mediumImageID)
            cell.episodeName.text = episode.name
            cell.seasonEpisode.text = "S\(episode.season) E: \(episode.number)"
            resultCell = cell
            
        }
        return resultCell